src/game/server/instagib/chat_commands.cpp

```diff
@@ -483,6 +483,56 @@ void CGameContext::ConRankFlagCaptures(IConsole::IResult *pResult, void *pUserDa
 	pSelf->m_pController->m_pSqlStats->ShowRank(pResult->m_ClientId, pName, "Flag captures", "flag_captures", pSelf->m_pController->StatsTable(), "DESC");
 }
 
+void CGameContext::ConTopSpikeColors(IConsole::IResult *pResult, void *pUserData)
+{
+	const auto *pSelf = static_cast<CGameContext *>(pUserData);
+	if(!CheckClientId(pResult->m_ClientId))
+		return;
+
+	if(!pSelf->m_pController)
+		return;
+
+	if(pSelf->m_pController->IsDDRaceGameType())
+	{
+		pSelf->SendChatTarget(pResult->m_ClientId, "This command is not available in ddrace gametypes.");
+		return;
+	}
+
+	const char *pName = pSelf->Server()->ClientName(pResult->m_ClientId);
+	const char *SpikeColor = pResult->GetString(0);
+	const int Offset = pResult->NumArguments() > 1 ? pResult->GetInteger(1) : 1;
+	const char *aSpikeColors[] = {
+		"gold",
+		"green",
+		"purple"};
+
+	for(const char *pColor : aSpikeColors)
+	{
+		if(strcmp(SpikeColor, pColor) == 0)
+		{
+			char aDisplayName[64];
+			str_format(aDisplayName, sizeof(aDisplayName), "%s spikes", pColor);
+
+			char aDbColumn[64];
+			str_format(aDbColumn, sizeof(aDbColumn), "%s_spikes", pColor);
+
+			pSelf->m_pController->m_pSqlStats->ShowTop(
+				pResult->m_ClientId, pName,
+				aDisplayName,
+				aDbColumn,
+				pSelf->m_pController->StatsTable(),
+				"DESC",
+				Offset);
+			return;
+		}
+	}
+	pSelf->SendChatTarget(pResult->m_ClientId, "~~~ Spike colors");
+	for(const char *pColor : aSpikeColors)
+	{
+		pSelf->SendChatTarget(pResult->m_ClientId, ("~ " + std::string(pColor)).c_str());
```

Comment:
Please don't use ``std::string`` use ``str_format`` instead to stay in the tw code style.